homeassistant/components/tolosauna/climate.py

```diff
@@ -0,0 +1,166 @@
+        )
+        self._attr_target_temperature_step = 1
+        self._attr_temperature_unit = TEMP_CELSIUS
+        self._attr_unique_id = f"{self._config_entry.entry_id}_climate"
+
+    @property
+    def current_temperature(self) -> int:
+        """Return current temperature."""
+        return self.status.current_temperature
+
+    @property
+    def current_humidity(self) -> int:
+        """Return current humidity."""
+        return self.status.current_humidity
+
+    @property
+    def target_temperature(self) -> int:
+        """Return target temperature."""
+        return self.settings.target_temperature
+
+    @property
+    def target_humidity(self) -> int:
+        """Return target humidity."""
+        return self.settings.target_humidity
+
+    @property
+    def hvac_mode(self) -> str:
+        """Get current HVAC mode."""
+        if self.status.power_on:
+            return HVAC_MODE_HEAT
+        if not self.status.power_on and self.status.fan_on:
+            return HVAC_MODE_DRY
+        return HVAC_MODE_OFF
+
+    @property
+    def hvac_action(self) -> str | None:
+        """Execute HVAC action."""
+        if self.status.calefaction == Calefaction.HEAT:
+            return CURRENT_HVAC_HEAT
+        if self.status.calefaction == Calefaction.KEEP:
+            return CURRENT_HVAC_IDLE
+        if self.status.calefaction == Calefaction.INACTIVE:
+            if self.status.fan_on:
+                return CURRENT_HVAC_DRY
+            return CURRENT_HVAC_OFF
+        return None
+
+    @property
+    def fan_mode(self) -> str:
+        """Return current fan mode."""
+        if self.status.fan_on:
+            return FAN_ON
+        return FAN_OFF
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set HVAC mode."""
+        if hvac_mode == HVAC_MODE_OFF:
+            await self.hass.async_add_executor_job(
+                self._set_power_and_fan, False, False
+            )
+        if hvac_mode == HVAC_MODE_HEAT:
+            await self.hass.async_add_executor_job(self._set_power_and_fan, True, False)
+        if hvac_mode == HVAC_MODE_DRY:
+            await self.hass.async_add_executor_job(self._set_power_and_fan, False, True)
+
+        await self.coordinator.async_refresh()
+
+    async def async_set_fan_mode(self, fan_mode: str) -> None:
+        """Set fan mode."""
+        if fan_mode == FAN_OFF:
+            await self.hass.async_add_executor_job(self.client.set_fan_on, False)
+        if fan_mode == FAN_ON:
+            await self.hass.async_add_executor_job(self.client.set_fan_on, True)
+
+        await self.coordinator.async_refresh()
+
+    async def async_set_humidity(self, humidity: float) -> None:
+        """Set desired target humidity."""
+        await self.hass.async_add_executor_job(
+            self.client.set_target_humidity, int(humidity)
+        )
+        await self.coordinator.async_refresh()
```

Comment:
```suggestion
    def set_humidity(self, humidity: float) -> None:
        """Set desired target humidity."""
        self.coordinator.client.set_target_humidity(int(humidity))
```

Additional thought: Does it need to be casted to an `int` or rounded to an `int`?

Reply:
I think it doesn't make a big difference, but I guess `round` is better. Thanks for the hint.